Extension/Services/TroopActionService.cs

```diff
@@ -78,25 +78,39 @@ public static (int, int) RecruitPrisoners(PartyVM partyVm, PartyScreenLogic part
                 }
             }
 
+            var playerSelectedCharacter = partyVm.CurrentCharacter.Character;
+
             int recruitedTypes = 0;
             int recruitedTotal = 0;
             List<Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>> commands = new List<Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>>();
-            while (recruitablePrisoners.Count > 0 && (partySpace > 0 || States.HotkeyControl))
+            foreach (var prisoners in recruitablePrisoners.OrderByDescending(vm => vm.Character.Tier))
             {
-                PartyCharacterVM prisoners = recruitablePrisoners.First();
+                if (recruitablePrisoners.Count == 0 || (partySpace == 0 && !States.HotkeyControl))
+                    break;
+
                 int prisonerCount = prisoners.NumOfRecruitablePrisoners;
                 int prisonersToRecruit = States.HotkeyControl ? prisonerCount : Math.Min(prisonerCount, partySpace);
+                int numOfRemainingRecruitablePrisoners = prisonerCount - prisonersToRecruit;
                 recruitedTypes++;
                 recruitedTotal += prisonersToRecruit;
                 partySpace -= prisonersToRecruit;
 
+                partyVm.CurrentCharacter.Character = prisoners.Character;
+
+                Campaign.Current.GetCampaignBehavior<IRecruitPrisonersCampaignBehavior>()
+                    .SetRecruitableNumber(partyVm.CurrentCharacter.Character, numOfRemainingRecruitablePrisoners + 1);
+
                 PartyScreenLogic.PartyCommand recruitCommand = new PartyScreenLogic.PartyCommand();
                 recruitCommand.FillForRecruitTroop(prisoners.Side, prisoners.Type, prisoners.Character, prisonersToRecruit);
                 commands.Add(new Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>(prisoners, recruitCommand));
 
+                partyVm.CurrentCharacter.UpdateRecruitable();
+
                 recruitablePrisoners.RemoveAt(0);
```

Comment:
If changing to a foreach, this will cause a runtime error as the collection is being edited. Either use foreach and remove, or keep while in place